test/preview/components/generators/openai/test_chatgpt_generator.py

```diff
@@ -0,0 +1,322 @@
+        component = ChatGPTGenerator()
+        data = component.to_dict()
+        assert data == {
+            "type": "ChatGPTGenerator",
+            "init_parameters": {
+                "api_key": None,
+                "model_name": "gpt-3.5-turbo",
+                "system_prompt": None,
+                "model_parameters": None,
+                "streaming_callback": None,
+                "api_base_url": "https://api.openai.com/v1",
+            },
+        }
+
+    @pytest.mark.unit
+    def test_to_dict_with_parameters(self):
+        component = ChatGPTGenerator(
+            api_key="test-api-key",
+            model_name="gpt-4",
+            system_prompt="test-system-prompt",
+            model_parameters={"max_tokens": 10, "some-test-params": "test-params"},
+            streaming_callback=default_streaming_callback,
+            api_base_url="test-base-url",
+        )
+        data = component.to_dict()
+        assert data == {
+            "type": "ChatGPTGenerator",
+            "init_parameters": {
+                "api_key": "test-api-key",
+                "model_name": "gpt-4",
+                "system_prompt": "test-system-prompt",
+                "model_parameters": {"max_tokens": 10, "some-test-params": "test-params"},
+                "api_base_url": "test-base-url",
+                "streaming_callback": "haystack.preview.components.generators.openai.chatgpt.default_streaming_callback",
+            },
+        }
+
+    @pytest.mark.unit
+    def test_from_dict(self):
+        data = {
+            "type": "ChatGPTGenerator",
+            "init_parameters": {
+                "api_key": "test-api-key",
+                "model_name": "gpt-4",
+                "system_prompt": "test-system-prompt",
+                "model_parameters": {"max_tokens": 10, "some-test-params": "test-params"},
+                "api_base_url": "test-base-url",
+                "streaming_callback": "haystack.preview.components.generators.openai.chatgpt.default_streaming_callback",
+            },
+        }
+        component = ChatGPTGenerator.from_dict(data)
+        assert component.system_prompt == "test-system-prompt"
+        assert component.api_key == "test-api-key"
+        assert component.model_name == "gpt-4"
+        assert component.streaming_callback == default_streaming_callback
+        assert component.api_base_url == "test-base-url"
+        assert component.model_parameters == {"max_tokens": 10, "some-test-params": "test-params"}
+
+    @pytest.mark.unit
+    def test_run_no_api_key(self):
+        component = ChatGPTGenerator()
+        with pytest.raises(ValueError, match="OpenAI API key is missing. Please provide an API key."):
+            component.run(prompts=["test"])
+
+    @pytest.mark.unit
+    def test_run_no_system_prompt(self):
+        with patch(
+            "haystack.preview.components.generators.openai.chatgpt.openai.ChatCompletion.create"
+        ) as chatgpt_patch:
```

Comment:
I'd patch the whole test function if we need to wrap the whole body in its context.

Reply:
Same for the tests below.